### PR TITLE
[WIP] Backports for 1.0.4

### PR DESCRIPTION
**Pull request by fredrikekre**

### CODE REVIEW

**Comment by fredrikekre** [comment]
bors try

**Comment by KristofferC** [comment]
Update Project file?

**Comment by fredrikekre** [comment]
bors try

File: src/Operations.jl [hunk 1]
```diff
@@ -1166,6 +1166,7 @@ function rm(ctx::Context, pkgs::Vector{PackageSpec})
     end
     deps_names = collect(keys(ctx.env.project["deps"]))
     if haskey(ctx.env.project, "targets")
+        append!(deps_names, keys(ctx.env.project["extras"]))
```
Comment (fredrikekre) [inline src/Operations.jl]: @00vareladavid (or anyone really) 27d452d73ed33ac368b3cbf6675a2eaca1ff39a9 did not backport cleanly, does this look like an OK backport?

Comment (00vareladavid) [reply src/Operations.jl]: Yeah, this seems OK. Doesn't it need a `collect` in there though?

**Pull request merged**